core/src/main/java/org/infinispan/CachePublisher.java

```diff
@@ -0,0 +1,222 @@
+package org.infinispan;
+
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+
+import org.infinispan.commons.util.Experimental;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.configuration.cache.StateTransferConfiguration;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.reactive.publisher.PublisherReducers;
+import org.infinispan.reactive.publisher.impl.DeliveryGuarantee;
+import org.infinispan.reactive.publisher.impl.SegmentPublisherSupplier;
+import org.infinispan.util.function.SerializableBiConsumer;
+import org.infinispan.util.function.SerializableBiFunction;
+import org.infinispan.util.function.SerializableFunction;
+import org.reactivestreams.Publisher;
```

Comment:
Should we use the JDK one instead of exposing RxJava?

Reply:
This is the reactive streams Publisher btw, which is not RxJava specific. Also the client currently exposes org.reactivestreams.Publisher but none of the others. The "new" API is the one that would have support for JDK Publisher.